builtin/logical/pki/path_acme_order.go

```diff
@@ -512,6 +512,13 @@ func (b *backend) acmeGetOrderHandler(ac *acmeContext, _ *logical.Request, field
 		return nil, err
 	}
 
+	if order.Status == ACMEOrderPending {
+		// Lets see if we can update our order status to ready if all the authorizations have been completed.
+		if requiredAuthorizationsCompleted(b, ac, uc, order) {
+			order.Status = ACMEOrderReady
```

Comment:
If it is ready, should we persist it so we don't have to recompute?

Reply:
I'm a little worried about persisting it here as we might race against a call to `acmeFinalizeOrderHandler`, which we would lose/orphan the generated certificate.

Reply:
Do we really want a write on a read call?

Reply:
Ah gotcha. Fair enough, its fairly cheap to recompute, other than fetching the additional authorizations.